### PR TITLE
To find stack name also match on instance id

### PR DESCRIPTION
**Pull request by lbodor**

### CODE REVIEW

File: gws-core/src/main/java/au/gov/ga/geodesy/support/aws/Aws.java [hunk 1]
```diff
@@ -17,14 +17,19 @@
...
-        return EC2MetadataUtils.getInstanceId() != null;
+        return getInstanceId() != null;
+    }
+
+    public static final String getInstanceId() {
+        return EC2MetadataUtils.getInstanceId();
     }
 
     public static Optional<String> getStackName() {
+        String instanceId = getInstanceId();
```
Comment (oehm-smith) [inline gws-core/src/main/java/au/gov/ga/geodesy/support/aws/Aws.java]: Does the compiler not complain that you need to access this as a static?

File: gws-core/src/main/java/au/gov/ga/geodesy/support/aws/Aws.java [hunk 2]
```diff
@@ -17,14 +17,19 @@
...
         if (inAmazon()) {
             DescribeTagsResult tags = ec2.describeTags();
             return tags.getTags().stream()
-                .filter(t -> t.getKey().equals("aws:cloudformation:stack-name"))
+                .filter(t -> instanceId.equals(t.getResourceId()) && "aws:cloudformation:stack-name".equals(t.getKey()))
```
Comment (oehm-smith) [inline gws-core/src/main/java/au/gov/ga/geodesy/support/aws/Aws.java]: To me this feels like your code is tightly coupled to expectation that there is a key with that name.

Comment (lbodor) [inline gws-core/src/main/java/au/gov/ga/geodesy/support/aws/Aws.java]: It is.

File: gws-core/src/main/java/au/gov/ga/geodesy/support/aws/Aws.java (hunk 1 above)
Comment (lbodor) [inline gws-core/src/main/java/au/gov/ga/geodesy/support/aws/Aws.java]: It doesn't.

**Pull request merged**